src/hera/task.py

```diff
@@ -237,6 +237,17 @@ def ip(self):
         """Returns the specifications for the IP property of the task"""
         return f'"{{{{tasks.{self.name}.ip}}}}"'
 
+    def _get_dependency_tasks(self) -> List[str]:
+        """Extract task names from `depends` string"""
+        if self.depends is None:
+            return []
+        # Filter out operators
+        all_operators = [o.value for o in Operator]
+        tasks = [t for t in self.depends.split() if t not in all_operators]
+        # Remove dot suffixes
+        task_names = [t.split(".")[0] for t in tasks]
```

Comment:
Is it possible for this to yield false positives via `dag.tasks.X` specifications?

Reply:
I _think_ it might if the user sets their own `self.depends` to something custom.